src/lib.rs

```diff
@@ -145,21 +145,45 @@ use crate::time::{OffsetDateTime, PrimitiveDateTime, UtcOffset};
 /// to prevent unintended usage.
 #[derive(Clone, Copy, PartialEq, Eq, PartialOrd, Ord)]
 pub struct DateTime {
-    unix_timestamp: i64,
+    // Timestamp in microseconds.
+    pub(crate) timestamp_micros: i64,
 }
 
 impl DateTime {
-    /// Create new from UNIX timestamp
-    pub const fn from_unix_timestamp(unix_timestamp: i64) -> Self {
-        Self { unix_timestamp }
+    /// Create new from UNIX timestamp in seconds
+    pub const fn from_timestamp_secs(seconds: i64) -> Self {
+        Self {
+            timestamp_micros: seconds * 1_000_000,
+        }
+    }
+
+    /// Create new from UNIX timestamp in milliseconds
+    pub const fn from_timestamp_millis(milliseconds: i64) -> Self {
+        Self {
+            timestamp_micros: milliseconds * 1_000,
+        }
+    }
+
+    /// Create new from UNIX timestamp in microseconds.
+    pub const fn from_timestamp_micros(microseconds: i64) -> Self {
+        Self {
+            timestamp_micros: microseconds,
+        }
+    }
+
+    /// Create new from UNIX timestamp in nanoseconds.
+    pub const fn from_timestamp_nanos(nanoseconds: i128) -> Self {
+        Self {
+            timestamp_micros: (nanoseconds as i64 / 1_000),
```

Comment:
We should divide into an `i128` first and then convert to `i64`. [See for yourself](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=f86402599d1d1ac71742503ffb37d0a9). Can you add the corresponding unit test?

Reply:
inlined it and added a test for it.